arangod/RestServer/LanguageCheckFeature.cpp

```diff
@@ -133,28 +179,46 @@ LanguageCheckFeature::LanguageCheckFeature(Server& server)
 }
 
 void LanguageCheckFeature::start() {
+  using namespace arangodb::basics;
+
   auto& feature = server().getFeature<LanguageFeature>();
-  auto defaultLang = feature.getDefaultLanguage();
-  auto language = feature.getCollatorLanguage();
-  auto previous = ::getOrSetPreviousLanguage(server(), language);
+  auto [currLang, currLangType] = feature.getLanguage();
+  auto collatorLang = feature.getCollatorLanguage();
+
+  auto [prevLang, prevLangType] =
+      ::getOrSetPreviousLanguage(server(), collatorLang, currLangType);
+
+  if (LanguageType::INVALID == currLangType) {
+    LOG_TOPIC("7ef61", FATAL, arangodb::Logger::CONFIG)
+        << "Specified language '" << collatorLang << " has invalid type";
+    FATAL_ERROR_EXIT();
+  }
 
-  if (defaultLang.empty() && !previous.empty()) {
-    // override the empty current setting with the previous one
-    feature.resetDefaultLanguage(previous);
+  if (currLang.empty() && LanguageType::DEFAULT == currLangType &&
+      !prevLang.empty()) {
+    // we found something in LANGUAGE file
+    // override the empty current setting for default with the previous one
+    feature.resetLanguage(prevLang, prevLangType);
     return;
   }
 
-  if (language != previous) {
+  if (collatorLang != prevLang || prevLangType != currLangType) {
```

Comment:
Maybe here should be `currLang`?
Also if it is, I suggest  to add `TRI_ASSERT(currLang == collatorLang);`

Reply:
No, we should use `prevLang`. 

`prevLang` is a language from file.  `collatorLang` is a language, which is defined by user

Reply:
> No, we should use prevLang.

My question about possible replace`collatorLang` by `currLang`.
I thought about it because you write `prevLangType != currLangType`